### PR TITLE
Fixing single letter variables in vsa,wfau,xmatch

### PR DESCRIPTION
**Pull request by sebaucillon**
issue #1572 (renaming single letter variables in vsa, wfau, xmatch, tested by @mfonism )
<s>issue #1217 (changing docs from DR1 to DR2, untested)</s>

### CODE REVIEW

File: docs/gaia/gaia.rst [hunk 1]
```diff
@@ -501,9 +501,12 @@ To obtain a list of the tables shared to a user type the following:
   tap_schema.columns
   tap_schema.schemas
   tap_schema.key_columns
-  gaiadr1.phot_variable_time_series_gfov
-  gaiadr1.ppmxl_neighbourhood
-  gaiadr1.gsc23_neighbourhood
+  #gaiadr1.phot_variable_time_series_gfov
```
Comment (bsipocz) [inline docs/gaia/gaia.rst:504]: this seems to be coming from an unrelated commit, please remove the commit

File: astroquery/xmatch/core.py [hunk 2]
```diff
@@ -187,12 +187,12 @@ def _parse_text(self, text):
         """
         header = text.split("\n")[0]
         colnames = header.split(",")
-        for cn in colnames:
-            if colnames.count(cn) > 1:
-                ii = 1
-                while colnames.count(cn) > 0:
-                    colnames[colnames.index(cn)] = cn + "_{ii}".format(ii=ii)
-                    ii += 1
+        for item in colnames:
```
Comment (bsipocz) [inline astroquery/xmatch/core.py:190]: I would suggest to use descriptive variable names, that is also part of the motivation for this cleanup. 
E.g. `item` --> `column` or `colname`
`ii` -> could either stay or call it `count`, but `it` is not better than `ii` at all.

File: astroquery/xmatch/core.py [hunk 3]
```diff
@@ -109,11 +109,11 @@ def query_async(self, cat1, cat2, max_distance, colRA1=None, colDec1=None,
 
         return response
 
-    def _prepare_sending_table(self, i, payload, kwargs, cat, colRA, colDec):
+    def _prepare_sending_table(self, it, payload, kwargs, cat, colRA, colDec):
```
Comment (bsipocz) [inline astroquery/xmatch/core.py:112]: `it` is not descriptive. Maybe use `cat_index`

File: astroquery/wfau/core.py [hunk 4]
```diff
@@ -286,10 +286,10 @@ def get_images_async(self, coordinates, waveband='all', frame_type='stack',
         if verbose:
             print("Found {num} targets".format(num=len(image_urls)))
 
-        return [commons.FileContainer(U, encoding='binary',
+        return [commons.FileContainer(Uobj, encoding='binary',
```
Comment (bsipocz) [inline astroquery/wfau/core.py:289]: use `url` instead of `Uobj`

**Review by bsipocz — CHANGES_REQUESTED** [review]

File: docs/gaia/gaia.rst (hunk 1 above)
Comment (bsipocz) [reply docs/gaia/gaia.rst:504]: OK, now reading your description it's clear that you added changes for two issues in one PR. 

The general the suggestion is to avoid this, keep PRs self contained around one topic (either multiple issues of a module when part of a bigger refactor/development, or one issue for multiple modules).

Either way, this change doesn't address the issue. The output in this example should be what the commands above give back, and there are no commented out lines in that output.

**Comment by bsipocz** [comment]
There seems to be some misunderstanding about the workflow. You don't need to open PRs in your own fork. You can but it is not necessary as you can and should open a PR to the main astroquery repo from a branch from your fork (I guess the issue was that once you made your branches correctly, you went to github.com/sebaucillon/astroquery rather than github.com/astropy/astroquery). 

So, for the PR to astroquery, it should be done from your feature branch rather than from master. Doing so will save you a lot of headaches, as well as will make it much easier to work on multiple PRs parallel, and keeping up with the code changing in the meantime. Here is some documentation about the workflow: https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#workflow and also a worked out example of a pull request: https://docs.astropy.org/en/latest/development/workflow/git_edit_workflow_examples.html#astropy-fix-example

But now, we can still fix up this PR, by squashing all the commits together, and next time you can work from a feature branch. Squashing is very similar to an interactive rebase, so I'll go with a rebase route. Some docs is here: https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#rebase-but-only-if-asked

In summary, first you check whether the main repo is added as a remote:

`git remote -v`. You should be able to see something like the following (`astropy` maybe called `upstream`, and `sebaucillon` as `origin`, depending on whether you followed the astropy guide or a different one to set up your local clone. both are fine, but in the example below I assume it's `astropy` and `sebaucillon`)

```
astropy	git@github.com:astropy/astroquery.git (fetch)
astropy	git@github.com:astropy/astroquery.git (push)
sebaucillon	git@github.com:sebaucillon/astroquery (fetch)
sebaucillon	git@github.com:sebaucillon/astroquery (push)
```

First sync the content
`git fetch --all`

Make a copy of your branch, just to be on the safe side and have a backup you can easily come back to is something goes wrong.

`git branch copy_master_before_rebase`

You start an interactive rebase:

`git rebase -i astropy/master`

This brings up an editor, with 8 commits. Change the first `pick` to `reword`, and change the rest to say `fixup` rather than `pick`. Comment out the Gaia docs one. It should look something like this:

```
reword a06df6f3 Rename single letter internal variables, xmatch.
#pick 78d61cdc Try update gaia docs with DR2, no test.
fixup e4c7127e Rename single-letter variables in wfau.
fixup 1438ee8a Rename single letter variable in vsa.
fixup 6557556a Rename single letter variables, xmatch, except units as u.
fixup 4098c586 Rename single letter variables, wfau, except units as u.
fixup 7b618361 Rename single letter variables, vsa, except units as u.
fixup 16c8b6f1 Fix error in wfau w.r.t. single letter variable units u.
```

Save and exit. A new editor window will open up, edit the commit message there to add `wfau and vsa` to the end of the message, as all your chanages will end up in one commit. Save and exit. It should success without issues, and with a message `Successfully rebased and updated detached HEAD.`.

Now, push it back to your remote. It has to be a force push as this is changing the history.

`git push sebaucillon HEAD:master --force`

You should be all set, and the PR will most likely ready to be merged.

**Review by bsipocz — APPROVED** [review]

**Comment by bsipocz** [comment]
Thank you @sebaucillon, this all looks good now.

**Comment by sebaucillon** [comment]
> Thank you @sebaucillon, this all looks good now.

Thank you @bsipocz . I was indeed a bit confused with the whole workflow (I actually had just one remote, origin, with the fetch address being the astropy/astroquery and the push address being my sebaucillon/astroquery clone). 

Your comments made things a lot more clear for me. Thanks again.
